### PR TITLE
Add remark-language-server package

### PR DESCRIPTION
**Pull request by remcohaszing**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

This provides language server support for remark based on https://github.com/unifiedjs/unified-language-server/pull/13

Initially I was going to create this as a separate project, but maybe it belongs in the remark core project. Regardless, I think it belongs in the remark organization and at least now it’s no longer just a local project on my machine. :)

### CODE REVIEW

**Review by ChristianMurphy — APPROVED** [review]
Thanks @remcohaszing!

File: packages/remark-language-server/readme.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+*   [Contribute](#contribute)
+*   [Sponsor](#sponsor)
+*   [License](#license)
+
+## What is this?
+
+This package is a [language server][] which can lint and format markdown files
+using remark.
+
+You can choose from the 150+ plugins that already exist or make your own.
```
Comment (Murderlon) [inline packages/remark-language-server/readme.md:29]: Perhaps a small clarification that it's just another layer but the config can be shared between the remark CLI and editor integration.
 
```suggestion
The language server can take in your [remark][] configuration file so you can use the settings and all the plugins as you're used to. 
```

**Review by Murderlon — COMMENTED** [review]
Nice!

I do have to say it feels a bit off to have the language server inside `remarkjs/remark` but on the other hand it's just a different way to use remark, like the node API, or the CLI.

File: packages/remark-language-server/readme.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+```sh
+npm install remark-language-server
+```
+
+## Use
+
+Usage of this package depends on your editor integration.
+
+Practical examples are coming soon.
```
[the remark this answers is not in the document]
Comment (remcohaszing) [reply packages/remark-language-server/readme.md:53]: These practical examples depend on the release of this package though, so this is a bit of a circular dependency right now.

File: packages/remark-language-server/readme.md (hunk 1 above)
Comment (remcohaszing) [reply packages/remark-language-server/readme.md:29]: I actually copied the readme from `remark-cli` and modified it. The original line comes from there, so I’m not sure if it should be removed here.

**Review by Murderlon — APPROVED** [review]

**Comment by wooorm** [comment]
You mentioned it but didn’t really provide arguments: what’s the reason it should go in this repo?

File: packages/remark-language-server/package.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "remark-language-server",
+  "version": "1.0.0",
+  "description": "A language server for markdown using remark",
```
Comment (wooorm) [inline packages/remark-language-server/package.json:4]:
```suggestion
  "description": "Language server for markdown using remark",
```

File: packages/remark-language-server/package.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  "homepage": "https://remark.js.org",
+  "repository": "https://github.com/remarkjs/remark/tree/main/packages/remark-language-server",
+  "bugs": "https://github.com/remarkjs/remark-language-serve/issues",
+  "bin": "remark-language-server.js",
+  "author": "Remco Haszing <remcohaszing@gmail.com>",
```
Comment (wooorm) [inline packages/remark-language-server/package.json:9]: Are you the copyright holder? Is this one of the packages (or does it include work of) originally other folks?

File: packages/remark-language-server/package.json [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  "license": "MIT",
+  "type": "module",
+  "files": [
+    "remark-language-server.js"
+  ],
+  "scripts": {
+    "format": "remark . -qfo && prettier . -w --loglevel warn && xo --fix",
+    "test": "tsc && npm run format"
```
Comment (wooorm) [inline packages/remark-language-server/package.json:17]: Other packages in this monorepo have different scripts I believe, probably match those?

File: packages/remark-language-server/package.json [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    "mdast",
+    "lsp",
+    "languageserver",
+    "language server",
+    "cli"
+  ],
+  "dependencies": {
+    "unified-language-server": "^1.0.0"
+  }
+}
```
Comment (wooorm) [inline packages/remark-language-server/package.json:40]: Sorting of fields should match other packages

File: packages/remark-language-server/readme.md (hunk 1 above)
Comment (wooorm) [reply packages/remark-language-server/readme.md:29]: Maybe remove this? The previous and next sentence seem to explain “What is this”. This line seems to be about plugins, which aren’t introduced or explained in surrounding prose.

File: packages/remark-language-server/readme.md [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+  <br>
+  <a href="https://opencollective.com/unified"><strong>You?</strong></a>
+  <br><br>
+</td>
+</tr>
+</table>
+
+## License
+
+[MIT][license] © [Titus Wormer][author]
```
Comment (wooorm) [inline packages/remark-language-server/readme.md:163]: author should match `package.json#author`

File: packages/remark-language-server/remark-language-server.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env node
+import {createUnifiedLanguageServer} from 'unified-language-server'
+
+process.title = 'remark-language-server'
```
Comment (wooorm) [inline packages/remark-language-server/remark-language-server.js:4]: TIL! Is this common? Language-server-specific?

File: packages/remark-language-server/tsconfig.json [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    "checkJs": true,
+    "declaration": true,
+    "lib": ["ES2020"],
+    "module": "ES2020",
+    "moduleResolution": "node",
+    "noEmit": true,
+    "skipLibCheck": true,
+    "strict": true,
+    "target": "ES2020"
+  }
```
Comment (wooorm) [inline packages/remark-language-server/tsconfig.json:15]: Should use the main config, e.g.: https://github.com/remarkjs/remark/blob/main/packages/remark-parse/tsconfig.json

**Comment by remcohaszing** [comment]
> You mentioned it but didn’t really provide arguments: what’s the reason it should go in this repo?

Mainly because I wanted to publish this code somewhere and I’m not sure if I have permission to create projects within the remark organization. I’m not sure if this belongs in the remark core project or in a separate project. It’s fine with me if this ends up in a separate repo.

**Comment by wooorm** [comment]
I think a separate project makes the most sense?

One of us can also create an empty project for you, if needed

**Comment by remcohaszing** [comment]
Closing in favor of https://github.com/remarkjs/remark-language-server/pull/1

**Pull request closed without merging**